packages/block/template.ts

```diff
@@ -0,0 +1,124 @@
+import { AbstractBlock, EditType, Hole } from './types';
+import type { Edit, VElement } from './types';
+
+const IS_VOID_ELEMENT =
+  /area|base|basefont|bgsound|br|col|command|embed|frame|hr|image|img|input|isindex|keygen|link|menuitem|meta|nextid|param|source|track|wbr/i;
+
+export const renderToTemplate = (
+  vnode: VElement,
+  edits: Edit[] = [],
+  path: number[] = [],
+): string => {
+  let props = '';
+  let children = '';
+  const current: Edit = {
+    path, // The location of the edit in in the virtual node tree
+    edits: [], // Occur on mount + patch
+    inits: [], // Occur before mount
+  };
+
+  for (let name in vnode.props) {
+    const value = vnode.props[name];
+    if (name === 'key' || name === 'ref' || name === 'children') {
+      continue;
+    }
+    if (name === 'className') name = 'class';
+    if (name === 'for') name = 'htmlFor';
+
+    // eslint-disable-next-line @typescript-eslint/prefer-string-starts-ends-with
+    if (name[0] === 'o' && name[1] === 'n') {
```

Comment:
We should do a performance benchmark on this. On Chrome `startsWith` should be the fastest. [Benchmark](https://perf.link/#eyJpZCI6Im9xa2lrdTdjNWh1IiwidGl0bGUiOiIiLCJiZWZvcmUiOiJjb25zdCBzdHIgPSAnb25DbGljayciLCJ0ZXN0cyI6W3sibmFtZSI6InN0YXJ0c1dpdGgiLCJjb2RlIjoic3RyLnN0YXJ0c1dpdGgoJ29uJyk7IiwicnVucyI6WzQzNTQwMDAsMTUzNjAwMCw1MDQxMDAwLDU4ODYwMDAsMTMxMjAwMCw0NDEwMDAwLDE5MzQwMDAsMzQxOTAwMCw1MTM2MDAwLDE2NzAwMCw0NjY0MDAwLDIyMzUwMDAsNTQyNzAwMCw0NTI3MDAwLDUwNjQwMDAsNjUyMDAwLDM0NTUwMDAsNDc4MDAwMCw1MjEzMDAwLDQ5NTAwMCw2MTI0MDAwLDIyODAwMCwxMDgwMDAwLDQxODEwMDAsODEwMDAwLDMxMzYwMDAsMzUyMzAwMCwzMjA1MDAwLDEzMDAwMCwxNDQ3MDAwLDIyMzAwMCwxOTUwMDAsODkwMDAwLDMyNjQwMDAsMzIyMTAwMCwyOTY2MDAwLDM0NzAwMDAsMTE1MDAwLDQyMTIwMDAsMjA5ODAwMCw1NDI4MDAwLDE2ODYwMDAsMjA0NDAwMCw3NTEwMDAsMTg5NDAwMCwxMDc0MDAwLDI4NzYwMDAsMjc4NzAwMCw1NDQ4MDAwLDI4MzkwMDAsNDQ2MDAwLDE1MzcwMDAsMTY4ODAwMCw1Njc4MDAwLDgxODAwMCwzMDMwMDAsMTg2NzAwMCwxNzAyMDAwLDI3MDAwMCwyNTE5MDAwLDU2NDIwMDAsMzY5MjAwMCw1NDI4MDAwLDU2NjIwMDAsMzE1MzAwMCw1NDI4MDAwLDU0MjgwMDAsNTQyODAwMCwyMDYxMDAwLDU4ODYwMDAsMjk4ODAwMCw2MDkwMDAsMjkyNDAwMCwxNTM4MDAwLDQyNjgwMDAsNDIzNzAwMCwxNDQwMDAsNDIyNzAwMCwxNDU5MDAwLDIxMTEwMDAsNTIzODAwMCw4MTkwMDAsNTQyODAwMCwzOTk5MDAwLDMxNTIwMDAsMjMwMDAsMzUwMDAwLDU0MjgwMDAsNjAwODAwMCwzNzQyMDAwLDU0MjgwMDAsNTQyODAwMCwyODQzMDAwLDMwMDAwMCwxNTczMDAwLDUwMjIwMDAsMTE3OTAwMCwzMzQzMDAwLDMzMzAwMDAsNDI2MDAwXSwib3BzIjoyOTMyNTIwfSx7Im5hbWUiOiJjaGFyQ29kZUF0IiwiY29kZSI6InN0ci5jaGFyQ29kZUF0KDApID09PSAxMTEgJiYgc3RyLmNoYXJDb2RlQXQoMSkgPT09IDExMDsiLCJydW5zIjpbNDA4NDAwMCwxMzMyMDAwLDQwNzAwMCw0MjIwMDAwLDMyNzcwMDAsMjYzNjAwMCwzNjI4MDAwLDUzMTkwMDAsNDkxMTAwMCwyNDg2MDAwLDgyNjAwMCwxNzY1MDAwLDMyNzcwMDAsMTM5MzAwMCwyNTk5MDAwLDMyNzYwMDAsMjM0MDAwLDQ3MzAwMCw0NDg0MDAwLDM1NjAwMDAsMjA5NTAwMCwzNjI5MDAwLDI1MzAwMCwxMzEzMDAwLDMyNzcwMDAsMzUxOTAwMCw0NTAzMDAwLDE2NDMwMDAsMTY0NTAwMCwzMjc3MDAwLDM4NjEwMDAsMjUxNTAwMCwzNDc0MDAwLDE3NTgwMDAsMTgxMTAwMCwxNDkwMDAwLDg2MzAwMCwzNDYzMDAwLDUwNjQwMDAsMzgwNzAwMCw0NzczMDAwLDkzMDAwLDE4MDAwLDE3MTYwMDAsNDEwMzAwMCw0MTE5MDAwLDE5ODcwMDAsMjk0ODAwMCwzMjc3MDAwLDE3NjAwMCwxNjk5MDAwLDQwMDAsMjA3NTAwMCwxNjIxMDAwLDM2MzMwMDAsMTg2MjAwMCwxMzg0MDAwLDI0NzgwMDAsMzMxMTAwMCw0MTkyMDAwLDMyNzcwMDAsMTUwMDAsNTU3MDAwLDI2MDgwMDAsMjUxNjAwMCwzMzYzMDAwLDE4NTMwMDAsMzQ4NDAwMCwzMjc2MDAwLDExMTMwMDAsMTM3MjAwMCwyMjQ5MDAwLDEyNDUwMDAsMjE3OTAwMCw1MjY3MDAwLDI2MDEwMDAsMjQ3MDAwMCw1MTQwMDAsNTIyOTAwMCwzNDczMDAwLDQwOTkwMDAsMzI3NzAwMCwzNDQxMDAwLDMxMDQwMDAsMjc3OTAwMCwxMDUxMDAwLDMzNDgwMDAsOTg2MDAwLDI4NjIwMDAsMzI3NzAwMCwzMjc3MDAwLDQzODAwMCwxMDk3MDAwLDI0MTcwMDAsNTUzMDAwLDMyNzcwMDAsNDUzNDAwMCwxMDUxMDAwLDE0MzAwMDAsMjk1MDAwMF0sIm9wcyI6MjUwODI1MH0seyJuYW1lIjoiWzBdIGFuZCBbMV0iLCJjb2RlIjoic3RyWzBdID09PSAnbycgJiYgc3RyWzFdID09PSAnbic7IiwicnVucyI6WzE5ODAwMDAsMTQ0NjAwMCwzMDcyMDAwLDI1MjEwMDAsNTYwNDAwMCw0MDUzMDAwLDQwNTMwMDAsMTEyMDAwMCw2MjYwMDAsMzE4MDAwMCwxOTI4MDAwLDQwNzgwMDAsMTcwMDAsNTQyNTAwMCwyOTc5MDAwLDI5NDAwMCwxODk1MDAwLDIxNDIwMDAsNjMwMDAsMzI4MzAwMCw0OTY2MDAwLDIyMTAwMCwzMTQzMDAwLDgyNDAwMCw1MzY4MDAwLDE3MzIwMDAsMjcyNTAwMCwyOTAwMCwyMDY4MDAwLDM0NDIwMDAsMjIwMTAwMCwzODMyMDAwLDUxMjkwMDAsNTIxMDAwMCw0OTE1MDAwLDY3NTAwMCw0Mjc2MDAwLDI4NzAwMCw0ODczMDAwLDIxNjMwMDAsMTcwMDAsMzg0NzAwMCwzMjIxMDAwLDM5MDcwMDAsMTQ0NjAwMCwzMTkzMDAwLDQ5NjAwMCw1Mzg5MDAwLDM1MDQwMDAsNDk1NjAwMCw0Mjc3MDAwLDQyNjMwMDAsNTI1ODAwMCw0MDUzMDAwLDQxMDIwMDAsMTc2OTAwMCw1NDAwMCwzMjQwMDAsNDA1MzAwMCwyOTcwMDAwLDE0NjgwMDAsMTgyMjAwMCwzNjczMDAwLDQwNTMwMDAsOTIyMDAwLDQ1NDYwMDAsNDg2MzAwMCw0MjAwMDAwLDM3OTkwMDAsNDA1MzAwMCwyOTM4MDAwLDg3MDAwMCw5MDQwMDAsMTA0ODAwMCwxODYxMDAwLDM1NzAwMDAsMTY1ODAwMCwyMTQ2MDAwLDM0NzkwMDAsMTE3NDAwMCw3OTQwMDAsMzE5ODAwMCwxNzUyMDAwLDU1NTAwMCw1MzYyMDAwLDMzNDUwMDAsMzU0NzAwMCw0MDUyMDAwLDQwNTMwMDAsMTMwNDAwMCwyNTg5MDAwLDMzNTUwMDAsMTUxNzAwMCw0MzgwMDAsNDA1MzAwMCwzMTU1MDAwLDQwNTIwMDAsNTQwMDAsMzEwMDAsNDM3NDAwMF0sIm9wcyI6MjczNTY0MH1dLCJ1cGRhdGVkIjoiMjAyMy0wMi0xNlQwOTowNjoxMy4yNjNaIn0%3D)

Reply:
fixed!

Reply:
Depending on the website, results aren't the same at all:

<img width="768" alt="Screenshot 2023-02-16 at 10 43 44" src="https://user-images.githubusercontent.com/43268759/219328526-823a66f8-3a14-458e-a79c-4f0d0bfd8de1.png">

<img width="828" alt="Screenshot 2023-02-16 at 10 43 54" src="https://user-images.githubusercontent.com/43268759/219328561-f2a15ab2-3fd7-4d1c-801c-f53b3242f149.png">

But agree on the benchmarks, we should have them on a repository so we can track the performance improvements in V8.

Reply:
> Depending on the website, results aren't the same at all:
>
> But agree on the benchmarks, we should have them on a repository so we can track the performance improvements in V8.

Yes, that's the point.
But I do prefer `startsWith` here. Not only is it fast, but also a few bytes smaller.